src/stream_p.cpp

```diff
@@ -120,25 +123,39 @@ void Aulib::Stream_priv::fSdlCallbackImpl(void* /*unused*/, Uint8 out[], int out
             continue;
         }
 
+        int diffMilliseconds = now - stream->d->fPlaybackStartTick;
+        if (diffMilliseconds <= 0) {
+            continue;
+        }
+
         bool has_finished = false;
         bool has_looped = false;
-        int len = 0;
+        int offset = 0;
+        int end = 0;
+
+        if (diffMilliseconds < bufferMilliseconds) {
+            int offsetMilliseconds = bufferMilliseconds - diffMilliseconds;
+            offset = wantedSamples * offsetMilliseconds / bufferMilliseconds;
```

Comment:
I believe this should be:

```
offset = offsetMilliseconds * fAudioSpec.channels * fAudioSpec.freq / 1000;
```

Using the ratio of the two millisecond values seems way too coarse and the resulting offset would end up being quite inaccurate.